test/SILOptimizer/mandatory_combiner.sil

```diff
@@ -75,6 +75,12 @@ sil @myint_from_myint_and_proto : $@convention(thin) (MyInt, @guaranteed { var P
 
 sil @myint_from_proto_and_myint : $@convention(thin) (@guaranteed { var Proto }, MyInt) -> MyInt
 
+// Optional support
+enum FakeOptional<T> {
```

Comment:
Looks like this declaration isn't actually used in the test...?

It would be good to test a _Swift_ program that uses this type, though. That's where we've run into problems with this optimization in the past.

Reply:
This is just misc boiler plate tests that I add in most tests. Its useful to have.